bots/mainbot/castle.js

```diff
@@ -170,4 +177,34 @@ castle.takeTurn = (self) => {
 
 };
 
-export default castle;
\ No newline at end of file
+// Counts bots castle can see.
+castle.countUnits = (self, visibleBots) => {
+    let bot = undefined;
+    // Zero the counts
+    self.castle_count = 0;
+    self.crusader_count = 0;
+    self.pilgrim_count = 0;
+    self.prophet_count = 0;
+    for (let i = 0; i < visibleBots.length; ++i){
+        bot = visibleBots[i];
+        // utilities.log(self, `Bot at ${i} has unit ${bot.unit} and castleTalk ${bot.castleTalk}`);
+        switch (bot.unit) {
+        	case SPECS.CASTLE:
```

Comment:
Missing churches. Should it be counted with castles, or as their own separate category?

Reply:
Yeah so this can only count units the castle can see. Is it useful to know how many churches are in range of the castle? Especially if we only build them if the pilgrims are too far from the castle?

Reply:
The castles have a very large vision range; far enough that the pilgrims might decide it's easier to build a church instead of running back to the caste. I can't think of a particular reason we need to know the number of churches in range, I just wanted to point out that they were mission just in case it was an oversight

Reply:
Stopped building them because I have their movement commented out and they were using resources. The castle rework adds the different strategies including crusader stuff.